dali-mqtt-daemon.py

```diff
@@ -171,23 +182,39 @@ def main_loop(driver, max_lamps, mqtt_server, mqtt_port, ha_prefix):
               "ha_discover_prefix": args.ha_discover_prefix,
               }
 
+    exception_raised = False
     try:
-        with open(args.config, 'r') as stream:
-            logger.debug("Loading configuration from <%s>", args.config)
-            config = yaml.load(stream)
-
-        MQTT_BASE_TOPIC = config["mqtt_base_topic"]
-
-        main_loop(config["dali_driver"], config["dali_lamps"], config["mqtt_server"], config["mqtt_port"], config["ha_discover_prefix"])
+        driver_object = None
+        driver = args.dali_driver
+        logger.debug("Using <%s> driver", driver)
+        if driver == HASSEB:
+            from dali.driver.hasseb import SyncHassebDALIUSBDriver 
+            driver_object = SyncHassebDALIUSBDriver()
+        elif driver == TRIDONIC:
+            from dali.driver.tridonic import SyncTridonicDALIUSBDriver
+            driver_object = SyncTridonicDALIUSBDriver()
+        elif driver == DALI_SERVER:
+            from dali.driver.daliserver import DaliServer
+            driver_object = DaliServer("localhost", 55825)    
+       
+        watchdog_observer = Observer()
+        watchdog_event_handler = ConfigFileSystemEventHandler()
+        watchdog_event_handler.on_modified = lambda event: on_detect_changes_in_config(event, watchdog_event_handler.mqqt_client)
+        watchdog_observer.schedule(watchdog_event_handler, args.config)
+        watchdog_observer.start()
+        main_loop(args.config, driver_object, watchdog_event_handler)
```

Comment:
main_loop is trivial now, best to move it's code to here

Reply:
Done